nav2_costmap_2d/plugins/denoise_layer.cpp

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2021 Andrey Ryzhikov
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_costmap_2d/denoise_layer.hpp"
+
+#include <string>
+#include <vector>
+#include <algorithm>
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_costmap_2d
+{
+bool isBackground(uint8_t pixel)
```

Comment:
Make a reference to reduce a copy.

Reply:
Are we talking about replacing `bool isBackground(uint8_t pixel)` with `bool isBackground(const uint8_t& pixel)`? But [it seems](https://godbolt.org/z/a1o3zsT3T) that passing primitives by value is more efficient.